### PR TITLE
Secrets as resources: more review feedback

### PR DESCRIPTION
**Pull request by benlangfeld**

### CODE REVIEW

File: test/integration/kubernetes_deploy_test.rb [hunk 1]
```diff
@@ -1098,6 +1098,46 @@ def test_network_policies_are_deployed_first
     ], in_order: true)
   end
 
+  # We want to be sure that failures to apply resources with potentially sensitive output don't leak any content.
+  # Currently our only sensitive resource is `Secret`, but we cannot reproduce a failure scenario where the kubectl
+  # output contains the filename (which would trigger some extra logging). This test stubs `Deployment` to be sensitive
+  # to recreate such a condition
+  def test_apply_failure_with_sensitive_resources_hides_template_content
+    logger.level = 0
+    KubernetesDeploy::Deployment.any_instance.expects(:kubectl_output_is_sensitive?).returns(true).at_least_once
```
Comment (KnVerey) [inline test/integration/kubernetes_deploy_test.rb]: This definitely cannot be run in parallel. `.any_instance.expects` is from `mocha`, which is not thread-safe

Comment (benlangfeld) [reply test/integration/kubernetes_deploy_test.rb]: Resolved.

File: lib/kubernetes-deploy/deploy_task.rb [hunk 2]
```diff
@@ -465,8 +465,8 @@ def record_apply_failure(err, resources: [])
         if bad_files.present?
           bad_files.each do |f|
             if sensitive_filenames.include?(f[:filename])
-              # Hide the template contents in case it has senitive information
-              record_invalid_template(err: f[:err], filename: f[:filename], content: nil)
+              # Hide the error and template contents in case it has senitive information
+              record_invalid_template(err: "SUPRESSED FOR SECURITY", filename: f[:filename], content: nil)
```
Comment (KnVerey) [inline lib/kubernetes-deploy/deploy_task.rb]: typo `SUPRESSED` -> `SUPPRESSED`

**Comment by KnVerey** [comment]
Triggered CI on `038ab18fc380589789eb176626ba8a6b9f4937a6`

**Comment by KnVerey** [comment]
Two failures:

```
Error:
KubernetesDeployTest#test_pruning_works:
ArgumentError: wrong number of arguments (given 1, expected 0)
    /usr/src/app/test/helpers/fixture_sets/hello_cloud.rb:119:in `assert_secret_present'
    /usr/src/app/test/helpers/fixture_sets/hello_cloud.rb:120:in `assert_secret_present'
    /usr/src/app/test/helpers/fixture_sets/hello_cloud.rb:24:in `assert_all_up'
    /usr/src/app/test/integration/kubernetes_deploy_test.rb:86:in `test_pruning_works'

Error:
KubernetesDeployTest#test_full_hello_cloud_set_deploy_succeeds:
ArgumentError: wrong number of arguments (given 1, expected 0)
    /usr/src/app/test/helpers/fixture_sets/hello_cloud.rb:119:in `assert_secret_present'
    /usr/src/app/test/helpers/fixture_sets/hello_cloud.rb:120:in `assert_secret_present'
    /usr/src/app/test/helpers/fixture_sets/hello_cloud.rb:24:in `assert_all_up'
    /usr/src/app/test/integration/kubernetes_deploy_test.rb:8:in `test_full_hello_cloud_set_deploy_succeeds'
```

**Comment by benlangfeld** [comment]
@KnVerey Apologies, was in a rush to comply with a review comment and that silly mistake slipped in. Resolved.

**Pull request merged**